### PR TITLE
Random maintenance rooms: Framework

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
(Note: none of this adds any real content to the game [yet].)
-Adds new subtypes to the random room spawner for six preset room sizes(3x3, 3x5, 5x3, 5x4, 10x5, 10x10), to make mapping them in easier
-adds a boxstation variant with random maint. Not to be used until all the random rooms are added. Do note only one version of each room can spawn, so most random room areas are just holes to space when the map is loaded.
-adds a template room for each size, to make mapping them easier
## Why It's Good For The Game

makes ike's random room template thing easier to use for mapping in the future, specifically for the purpose of random maint rooms

## Changelog
:cl:
add: 6 new random room spawner subtypes
add: random maint room templates
add: new random maint box station map for testing random rooms (note: does not have a 10x5 room)
/:cl:

### CODE REVIEW

**Comment by qwertyquerty** [comment]
What a mess of commits.

Anyway the template maps and template paths don't seem to serve any purpose.

**Comment by zeskorion** [comment]
> What a mess of commits.
> 
> Anyway the template maps and template paths don't seem to serve any purpose.

yeah, accidentally commited a few things to master awhile back. The template maps and paths are largely for testing and providing example to mappers until we have sufficient maps, after which they can be deleted

**Comment by qwertyquerty** [comment]
Nah, delete the templates, and their classes. Uneeded. Will merge when done

**Comment by zeskorion** [comment]
> Nah, delete the templates, and their classes. Uneeded. Will merge when done

nah, the template's classes are needed, because they organize this shit by preset sizes instead of having a jumbled mess. i can delete the templates themselves though

File: code/modules/mapping/random_rooms.dm [hunk 1]
```diff
@@ -4,6 +4,7 @@
 	var/centerspawner = TRUE
 	var/template_height = 0
 	var/template_width = 0
+//5x4
```
Comment (qwertyquerty) [inline code/modules/mapping/random_rooms.dm]: you forgot to remove this comment

**Comment by qwertyquerty** [comment]
Why do we have new map files for the random maintenance maps?

**Comment by zeskorion** [comment]
> Why do we have new map files for the random maintenance maps?

Forgot to make a new branch for that one

**Comment by qwertyquerty** [comment]
Boxstation one is still there, also you didn't fix the comment

**Comment by zeskorion** [comment]
> Boxstation one is still there, also you didn't fix the comment

It's there as a proof of concept and an easy way for mappers to test. But if you don't want it in there yet, fine by me

**Comment by qwertyquerty** [comment]
Yes please remove box

**Pull request merged**